apps/repo-server/src/microservices/prepareImageHTTPEndpoint.ts

```diff
@@ -0,0 +1,106 @@
+import bmpDecode from "@vingle/bmp-js";
+import heicDecode from "heic-decode";
+import { StatusCodes } from "http-status-codes";
+import sharp from "sharp";
+
+import { isIPrepareImageTaskArgs } from "@/tsrc/apps/repo-server/interface/IPrepareImageTaskArgs";
+import { uploadFile } from "~/apps/desktop-app/src/utils/uploadFile";
+import { sliceBufferAndCopyToNewArrayBuffer } from "~/apps/repo-server/src/sliceBufferAndCopyToNewArrayBuffer";
+import type { IHTTPEndpointArgs } from "~/lib/interface/IHTTPEndpointArgs";
+import type { IHTTPEndpointReturnType } from "~/lib/interface/IHTTPEndpointReturnType";
+import { LOG_LEVEL, Logger } from "~/lib/logger/Logger";
+
+export async function prepareImageHTTPEndpoint(
```

Comment:
Have you considered extracting the core part of this function to a class or function for which tests can be more easily written?

Reply:
I've added tests